tidy3d/components/geometry/mesh.py

```diff
@@ -57,35 +57,71 @@ def _warn_if_none(cls, val: TriangleMeshDataset) -> TriangleMeshDataset:
         return val
 
     @pydantic.validator("mesh_dataset", always=True)
+    @verify_packages_import(["trimesh"])
     def _check_mesh(cls, val: TriangleMeshDataset) -> TriangleMeshDataset:
         """Check that the mesh is valid."""
         if val is None:
             return None
+
+        import trimesh
+
         mesh = cls._triangles_to_trimesh(val.surface_mesh)
         if not all(np.array(mesh.area_faces) > AREA_SIZE_THRESHOLD):
-            raise ValidationError(
+            old_tol = trimesh.tol.merge
+            trimesh.tol.merge = np.sqrt(2 * AREA_SIZE_THRESHOLD)
+            new_mesh = mesh.process(validate=True)
+            trimesh.tol.merge = old_tol
+            val = TriangleMesh.from_trimesh(new_mesh).mesh_dataset
+            log.warning(
                 f"The provided mesh has triangles with near zero area < {AREA_SIZE_THRESHOLD}. "
-                "Consider using numpy-stl's 'from_file' import with 'remove_empty_areas' set "
-                "to True and a suitable 'AREA_SIZE_THRESHOLD' to remove them."
+                "These triangles are being automatically removed."
```

Comment:
Perhaps more precise warning: "Triangles which have one edge of their 2D oriented bounding box shorter than sqrt(2*threshold) are being automatically removed"